lib/bundlex/helper/mix_helper.ex

```diff
@@ -24,11 +26,41 @@ defmodule Bundlex.Helper.MixHelper do
     end
   end
 
-  def get_mixfile_env(application) do
-    case Application.get_env(:bundlex, application) do
-      %Macro.Env{} = env -> {:ok, env}
-      nil -> {:error, {:mixfile_env_undefined, application}}
-      invalid_env -> {:error, {:mixfile_env_invalid, application, invalid_env}}
+  @doc """
+  Stores current project directory in an agent.
+  This is necessary because `Mix.Project.deps_paths/0` sometimes return invalid
+  paths when dependencies are given by path
+  (see https://github.com/elixir-lang/elixir/issues/7561). Such dependencies are always
+  recompiled, thus proper path is always stored and can be retrieved by dependent
+  projects.
+  """
+  def store_project_dir() do
+    Agent.start(fn -> %{} end, name: @path_store_name)
+
+    with {:ok, dir} <- get_project_dir() do
+      Agent.update(@path_store_name, &(&1 |> Map.put(get_app!(), dir)))
```

Comment:
Is this pipe really needed? Without it you could skip parentheses and the line will be 4 characters shorter! IMHO readability would benefit as well :wink: